### ISSUE TITLE
--upload-only flag seems to be ignored for dev sync

### ISSUE THREAD

**Issue by gbates101** [devspace-cloud/devspace#1004]
**What happened?**  
When running `dev sync --config=./devspace.yaml --upload-only=true`, files from the pod are still downloaded prior to uploads being processed. I've also attempted running using `--upload-only` without the assignment, same outcome.

**What did you expect to happen instead?**  
Do not download any files if that flag is specified.

**How can we reproduce the bug?** (as minimally and precisely as possible)  
1. Run `devspace deploy` on any project.
2. (Re)move a file locally that matches a sync rule.
3. Start `devspace sync --config=./devspace.yaml --upload-only=true` from the root of your project.
4. Note the file (re)moved in step 2 is downloaded.

**Local Environment:**  
- DevSpace Version: 4.6.0
- Operating System: Windows (WSL)
- Deployment method: helm

**Kubernetes Cluster:**  
- Cloud Provider: google
- Kubernetes Version:
    - Client: v1.15.5
    - Server v1.14.8-gke.33

**Anything else we need to know?**  
Here's the relevant config section:
```
dev:
  sync:
  - imageName: plaidcloud/superset
    labelSelector:
      plaid: superset
    containerName: "superset"
    localSubPath: ./superset
    containerPath: /app/superset
  - imageName: plaidcloud/superset-node
    labelSelector:
      plaid: superset
    containerName: "node"
    localSubPath: ./superset-frontend/src
    containerPath: /app/superset-frontend/src
```
Here's a screenshot of the output:
![image](https://user-images.githubusercontent.com/24995887/75999263-f7fcc300-5ecf-11ea-80c7-ae6edd1e5941.png)
And another for the second rule:
![image](https://user-images.githubusercontent.com/24995887/75998360-b881a700-5ece-11ea-9179-d8f4451dabf4.png)

/kind bug

**Comment by LukasGentele** [comment]
@gbates101 Thanks for reporting this.

`--upload-only` should probably automatically set `--initial-sync=mirrorLocal`. Otherwise this behavior is very unexpected. What do you think, @FabianKramm?

**Comment by FabianKramm** [comment]
@gbates101 yes currently the flags `--upload-only` and `--download-only` are not respected when using in conjunction with `--config`. As a workaround you can either set `dev.sync.disableDownload` to true or use the `devspace sync` command without the `--config`

**Comment by gbates101** [comment]
@FabianKramm Ah, good to know. `dev.sync.disableDownload` is exactly what I'm looking for.

**Comment by FabianKramm** [comment]
implemented in v4.8.0
